pkg/golinters/gci.go

```diff
@@ -1,96 +1,43 @@
 package golinters
 
 import (
-	"bytes"
 	"fmt"
-	"sync"
+	"strings"
 
-	"github.com/daixiang0/gci/pkg/gci"
-	"github.com/pkg/errors"
-	"github.com/shazow/go-diff/difflib"
+	gci "github.com/daixiang0/gci/pkg/analyzer"
 	"golang.org/x/tools/go/analysis"
 
+	"github.com/golangci/golangci-lint/pkg/config"
 	"github.com/golangci/golangci-lint/pkg/golinters/goanalysis"
-	"github.com/golangci/golangci-lint/pkg/lint/linter"
 )
 
 const gciName = "gci"
 
-func NewGci() *goanalysis.Linter {
-	var mu sync.Mutex
-	var resIssues []goanalysis.Issue
-	differ := difflib.New()
+func NewGci(settings *config.GciSettings) *goanalysis.Linter {
+	var linterCfg map[string]map[string]interface{}
 
-	analyzer := &analysis.Analyzer{
-		Name: gciName,
-		Doc:  goanalysis.TheOnlyanalyzerDoc,
-	}
-	return goanalysis.NewLinter(
-		gciName,
-		"Gci control golang package import order and make it always deterministic.",
-		[]*analysis.Analyzer{analyzer},
-		nil,
-	).WithContextSetter(func(lintCtx *linter.Context) {
-		localFlag := lintCtx.Settings().Gci.LocalPrefixes
-		goimportsFlag := lintCtx.Settings().Goimports.LocalPrefixes
-		if localFlag == "" && goimportsFlag != "" {
-			localFlag = goimportsFlag
+	if settings != nil {
+		cfg := map[string]interface{}{
+			gci.NoInlineCommentsFlag:  settings.NoInlineComments,
+			gci.NoPrefixCommentsFlag:  settings.NoPrefixComments,
+			gci.SectionsFlag:          strings.Join(settings.Sections, gci.SectionDelimiter),
```

Comment:
We can use the code included in gci for backwards compatibility. I wanted to avoid direct dependencies on gci code as much as possible but I did not consider maintaining backwards compatibility when implementing the analyzer. So I guess this is the best compromise?

```suggestion
		sections := settings.Sections
		if settings.LocalPrefixes != "" && len(sections) == 0 {
			sections = actualGci.LocalFlagsToSections(strings.Split(settings.LocalPrefixes, ",")).String()
		}

		cfg := map[string]interface{}{
			gci.NoInlineCommentsFlag:  settings.NoInlineComments,
			gci.NoPrefixCommentsFlag:  settings.NoPrefixComments,
			gci.SectionsFlag:          strings.Join(sections, gci.SectionDelimiter),
```

Reply:
I prefer to handle the compatibility inside golangci-lint, like that we don't have to care about changes inside the linter.